sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubProducer/EventHubProducerLiveTests.cs

```diff
@@ -181,5 +186,649 @@ await using (var producer = client.CreateProducer())
+                                }
+                            }
+                        }
+                        finally
+                        {
+                            foreach (var consumer in consumers)
+                            {
+                                consumer.Close();
+                            }
+                        }
+
+                        Assert.That(partitionsCount, Is.GreaterThan(1));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubProducer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProducerSendsEventsInTheSameBatchToTheSamePartition()
+        {
+            var partitions = 10;
+
+            await using (var scope = await EventHubScope.CreateAsync(partitions))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                await using (var producer = client.CreateProducer())
+                {
+                    var eventBatch = Enumerable
+                        .Range(0, 30)
+                        .Select(index => new EventData(Encoding.UTF8.GetBytes("I'm getting used to this amount of messages")));
+
+                    var partitionIds = await client.GetPartitionIdsAsync();
+                    var partitionsCount = 0;
+                    var receivedEventsCount = 0;
+                    var consumers = new List<EventHubConsumer>();
+
+                    try
+                    {
+                        for (var index = 0; index < partitions; index++)
+                        {
+                            consumers.Add(client.CreateConsumer("$Default", partitionIds[index], EventPosition.Latest));
+
+                            // Initiate an operation to force the consumer to connect and set its position at the
+                            // end of the event stream.
+
+                            await consumers[index].ReceiveAsync(1, TimeSpan.Zero);
+                        }
+
+                        // Send the batch of events.
+
+                        await producer.SendAsync(eventBatch);
+
+                        // Receive the events; because there is some non-determinism in the messaging flow, the
+                        // sent events may not be immediately available.  Allow for a small number of attempts to receive, in order
+                        // to account for availability delays.
+
+                        foreach (var consumer in consumers)
+                        {
+                            var receivedEvents = new List<EventData>();
+                            var index = 0;
+
+                            while (++index < 5)
```

Comment:
Maybe we'll need to increase this value. This test failed once for me (out of aprox. 8 runs). It got only 4 messages instead of 30.

Reply:
I'm good with that;  the Event Hubs instance that we're using for Live tests is not a high-throughput SKU.    With the additional traffic around parallel test execution and management operations, it's not unlikely that we'll see some timing issues until we get the right thresholds in place.

I'd say, let's bump to `10` since we're only waiting 25 milliseconds for each attempt.   That gives the overall operation about 1/4 of a second.   If we need to, we can tune the wait up to 50ms if we still see inconsistency.   

Do you mind applying that across the Producer/Consumer tests?

Reply:
Done in Producer. I'll update it in Consumer in the next few commits.